File: 백준/Silver/1931. 회의실 배정/회의실 배정.swift
import Foundation

let N = Int(readLine()!)!
var meetings: [(Int, Int)] = []
for i in 0..<N {
    let n = readLine()!.split(separator: " ").map{Int(String($0))!}
    meetings.append((n[0], n[1]))
}

// 끝나는 시간이 가장 빠른 시간 순으로, 끝나는 시간이 같을 경우 시작 시간이 빠른 경우부터 정렬
meetings.sort{$0.1 == $1.1 ? $0.0 > $1.0 : $0.1 > $1.1}

var time = 0
var count = 0
while !meetings.isEmpty {
    let target = meetings.removeLast()
    if target.0 >= time {
        time = target.1
        count += 1
    }
}
print(count)

File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Main {

	public static void main(String[] args) throws NumberFormatException, IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		int[][] meetings = new int[N][2];
		for (int i = 0; i < N; i++) {
			meetings[i] = Arrays.stream(br.readLine().split(" ")).mapToInt(Integer::parseInt).toArray();
		}
		Arrays.sort(meetings, (o1, o2) ->  {
			if (o1[1] == o2[1]) return o1[0] - o2[0];
			return o1[1] - o2[1];
		});
		int count = 1;
		int currentFinishTime = meetings[0][1];
		
		for (int i = 1; i < N; i++) {
			if (currentFinishTime <= meetings[i][0]) {
				currentFinishTime = meetings[i][1];
				count++;
			}
		}
		System.out.println(count);
	}
}